frontend/lib/src/WidgetStateManager.ts

```diff
@@ -709,6 +713,32 @@ export class WidgetStateManager {
       this.props.formsDataChanged(this.formsData)
     }
   }
+
+  public setExtraWidgetInfo(
+    widget: WidgetInfo,
+    key: string,
+    value: any
+  ): void {
+    if (!this.extraWidgetInfo.has(widget.id)) {
+      this.extraWidgetInfo.set(widget.id, new Map<string, any>())
+    }
+
+    const currentMap = this.extraWidgetInfo.get(widget.id)
+
+    if (currentMap) {
+      currentMap.set(key, value)
+    }
+  }
+
+  public getExtraWidgetInfo(widget: WidgetInfo, key: string): any {
+    if (this.extraWidgetInfo.has(widget.id)) {
```

Comment:
Similar comment here: does this `if` statement do anything? If it weren't here and `this.extraWidgetInfo.get(widget.id)` returned undefined, then the second if statement simply wouldn't be executed and we'd return `undefined`, which is what we want